panther_diagnostics/include/panther_diagnostics/system_status.hpp

```diff
@@ -30,40 +32,37 @@ namespace panther_diagnostics
 class SystemStatus : public rclcpp::Node
 {
 public:
-  SystemStatus();
-
-  static constexpr const char * cpu_info_filename = "/proc/stat";
-  static constexpr const char * memory_info_filename = "/proc/meminfo";
-  static constexpr const char * temperature_info_filename = "/sys/class/thermal/thermal_zone0/temp";
+  SystemStatus(const std::string & node_name);
 
-  static constexpr float disk_usage_warn_threshold = 95.0;
-  static constexpr float memory_usage_warn_threshold = 95.0;
-  static constexpr float cpu_usage_warn_threshold = 95.0;
-  static constexpr float cpu_temp_warn_threshold = 80.0;
+  static constexpr const char * kCPUInfoFilename = "/proc/stat";
+  static constexpr const char * kMemoryInfoFilename = "/proc/meminfo";
+  static constexpr const char * kTemperatureInfoFilename = "/sys/class/thermal/thermal_zone0/temp";
 
 protected:
-  void TimerCallback();
-  float GetTemperature(const std::string & filename) const;
+  float GetCPUTemperature(const std::string & filename);
   std::vector<float> GetCPUsUsages(const std::string & filename);
-  float GetMemoryUsage(const std::string & filename) const;
-  float GetCPUMeanUsage() const;
-  float GetDiskUsage() const;
+  float GetMemoryUsage(const std::string & filename);
+  float GetCPUMeanUsage();
+  float GetDiskUsage();
+
   void ReadOneCPU(std::ifstream & file, const std::size_t index);
```

Comment:
The name provides us with much information about what the function is responsible for.

I propose:
- change return type void->float
- name this function somehow descriptive e.g. GetCoreUsage or GetCPUCoreUsage